### PR TITLE
Added home page, useAxios hook & Loader

### PR DESCRIPTION
**Pull request by kjais1720**
# PR for Issue #3 
## Overview
- Added useAxios hook to fetch data from API
- Added Loader component
- Added home page (hero banner and categories section only )

## Preview

screenshot (4)

### CODE REVIEW

File: src/pages/home/home.module.css [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    position: absolute;
+    inset: 0;
+    background-color: var(--tr-gray-overlay);
+    background-image: var(--bg-gradient-overlay);
+}
+
+.hero >*{
+    position: relative;
+    z-index: 1;
+} */
```
Comment (rahulcs754) [inline src/pages/home/home.module.css]: remove unnecessary comment from code

File: src/pages/home/home.module.css [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+.heroHeading{
+    font-size: clamp(2rem, 10vw - 1rem, 3rem);
+    color: var(--tr-background);
+    max-width: 30ch;
+}
+
+.cta{
+    font-size: 1.2rem;
+
```
Comment (rahulcs754) [inline src/pages/home/home.module.css:26]: try to remove space

File: src/components/loaders/loader.module.css [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+.loaderCircle2 {
+	animation-delay: 0.45s;
+}
+
+.loaderCircle3 {
+	animation-delay: 0.9s;
+}
+
+
+
```
Comment (rahulcs754) [inline src/components/loaders/loader.module.css]: remove space

**Review by rahulcs754 — APPROVED** [review]
All code is good and clean and liked you to create hooks for Axios function.
just small suggestion

File: src/pages/home/components/banner-section.jsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+export const HeroBanner = ({styles}) => {
+    return(
+        <section className={`${styles.hero} flex-col justify-c-center`} style={{backgroundImage:"url(/assets/hero-banner-2.webp),var(--bg-gradient-overlay)"}}>
```
Comment (coldpigli) [inline src/pages/home/components/banner-section.jsx:3]: Inline Styles ?

File: src/utilities/useAxios.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import {
+    useEffect,
+    useState
+} from "react";
+import axios from 'axios';
+
+/**
+ * 
+ * @param apiUrl : api url string
```
Comment (coldpigli) [inline src/utilities/useAxios.js:9]: Like how you are documenting function parameters.

**Review by coldpigli — APPROVED** [review]
Every thing looks really good.
Just found inline styling in a tag.

What I liked about your code:
1) Use of custom hooks for fetching data.
2) Documenting function parameters (It has motivated me to do the same).
3) Modular CSS.

File: src/pages/home/home.module.css (hunk 1 above)
Comment (kjais1720) [reply src/pages/home/home.module.css]: Thanks for pointing it out, fixed it in the latest commit

File: src/pages/home/home.module.css (hunk 2 above)
Comment (kjais1720) [reply src/pages/home/home.module.css:26]: Thanks for pointing it out, I have fixed it in the latest commit

File: src/pages/home/components/banner-section.jsx (hunk 4 above)
Comment (kjais1720) [reply src/pages/home/components/banner-section.jsx:3]: Actually, I wasn't able to access that image from the CSS file, so only way was to link in the js file

**Pull request merged**